### DISCUSSION TITLE
Can Supabase realtime watch database views?

### DISCUSSION THREAD

**Discussion by ThatXliner** [supabase/supabase#20099] [Questions] [answered]
It doesn't appear to work for me.

**Comment by GaryAustin1** [comment, marked as answer]
No. It only works on tables. 
You would need to monitor the table and then in the payload handler select from the view.

**Reply by ThatXliner** [reply]
Woah! Thanks for the fast response. Then is there a way to filter the realtime feed using a join (i.e. a column from another table)? In my testing, it doesn’t seem to work either. Or is that another thing I need to do manually?

Also, is the realtime feed going to be filtered through RLS?

**Reply by GaryAustin1** [reply]
No, there is no ability to do joins in the realtime data returned.  You would need to use the data returned from monitoring one or more tables with the basic filters offered, and then do a select to your view or do the joins there.  This would be done in the payload handler.

Realtime will use RLS for insert and update but not delete.   It uses the select RLS policy for everything.  For delete you get all deletes to the table (with the filter if any) but by default only get the primary key for the row for security.
